### PR TITLE
Lb/add sentry setup

### PR DESCRIPTION
**Pull request by elceebee**
## Context

We want to have a sentry integration so that we are aware of things when they go wrong. 

## Changes proposed in this pull request

- Add relevant sentry gems
- Sentry configuration

### Not in this PR, but related:
- [itt-mentor-services](https://dfe-teacher-services.sentry.io/settings/projects/itt-mentor-services/) has been set up as a project in the becoming-a-teacher 'team' in sentry
- Everyone has been added as a user and admin of the becoming-a-teacher team
- SENTRY_DNS environment variables have been added to review and qa -- we'll need to add it to the higher environments when they become available. 

## Guidance to review

- Login to the rails console for this review app (things have changed a bit since we all did this, so there is some new documentation coming, see Ollie's PR https://github.com/DFE-Digital/itt-mentor-services/pull/226)
- Manually send something to sentry, eg `Sentry.capture_message "This is a test"`
- Go to the [issues log for the project on Sentry](https://dfe-teacher-services.sentry.io/issues/?project=4506705538056192&referrer=sidebar&statsPeriod=14d)
- You should see the message you've sent.
- Please resolve the issue so we don't have issues hanging around unresolved. 

## Link to Trello card

https://trello.com/c/DFJaBFsv

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

## Screenshots

image

### CODE REVIEW

File: config/initializers/sentry.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Sentry.init do |config|
+  # Sentry config here is mostly copied from apply-for-teacher-training
+  # https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/initializers/sentry.rb
+  config.environment = ActiveSupport::EnvironmentInquirer.new(ENV.fetch("HOSTING_ENV", "development"))
```
Comment (ollietreend) [inline config/initializers/sentry.rb:4]: I think this should work? 🤔

```suggestion
  config.environment = HostingEnvironment.env
```

**Review by ollietreend — COMMENTED** [review]
Looks good to me! Just one thing.

File: config/initializers/sentry.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  config.inspect_exception_causes_for_exclusion = true
+
+  config.excluded_exceptions += [
+    # The following exceptions are user-errors that aren't actionable, and can be
+    # safely ignored.
+    "ActionController::BadRequest",
+    "ActionController::UnknownFormat",
+    "ActionController::UnknownHttpMethod",
+    "ActionDispatch::Http::Parameters::ParseError",
+    "Mime::Type::InvalidMimeType",
```
Comment (CatalinVoineag) [inline config/initializers/sentry.rb:17]: I think we can still get some value from some of these errors. If we have a lot of `BadRequest` it still might tell us something, maybe our form is not right or something else 🤷🏻

File: Gemfile [hunk 3]
```diff
@@ -79,6 +79,9 @@ gem "stimulus-rails"
 # User-Resource Permissions and Scoping
 gem "pundit"
 
+gem "sentry-rails"
+gem "sentry-ruby"
```
Comment (Nitemaeric) [inline Gemfile:83]: Do we need to explicitly add `sentry-ruby`? I think it's already a dependency for `sentry-rails`. :)

**Review by ollietreend — APPROVED** [review]

**Pull request merged**